### ISSUE TITLE
Unable to specify absolute path to report location on Linux

### ISSUE THREAD

**Issue by tillig** [LightBDD/LightBDD#137]
#### Description
Using .NET Core on Linux, I'm trying to use the ability to write reports by modifying `LightBddConfiguration` and specifying a report location. It looks like:

```csharp
configuration
  .ReportWritersConfiguration()
  .AddFileWriter<PlainTextReportFormatter>(
    "/out/{TestDateTimeUtc:yyyy-MM-dd-HH_mm_ss}_features.txt")
```
In doing that, it appeared that no reports at all were being generated. It was only after digging a bit that I found the reports were being generated... in the `bin/Debug/netcoreapp2.1/out` folder relative to my `.csproj`.

Here's how that happens:

When the [`ReportFileWriter` calculates the output path](https://github.com/LightBDD/LightBDD/blob/b2f4c1d886bba12f1998bafc64ebc356817eb354/src/LightBDD.Framework/Reporting/ReportFileWriter.cs#L51) it calls `ReportPathFormatter.ToFormattablePath` which [uses the `FilePathHelper`](https://github.com/LightBDD/LightBDD/blob/b2f4c1d886bba12f1998bafc64ebc356817eb354/src/LightBDD.Framework/Reporting/ReportPathFormatter.cs#L67) to [resolve the absolute path](https://github.com/LightBDD/LightBDD/blob/b2f4c1d886bba12f1998bafc64ebc356817eb354/src/LightBDD.Framework/Reporting/Implementation/FilePathHelper.cs#L8) for the report file.

However, the logic in there in attempting to resolve the absolute path [does some directory combining with trimming of path characters](https://github.com/LightBDD/LightBDD/blob/b2f4c1d886bba12f1998bafc64ebc356817eb354/src/LightBDD.Framework/Reporting/Implementation/FilePathHelper.cs#L29) that doesn't work on Linux.

The real key is in [`LightBDD.Framework.Reporting.Implementation.FilePathHelper`](https://github.com/LightBDD/LightBDD/blob/b2f4c1d886bba12f1998bafc64ebc356817eb354/src/LightBDD.Framework/Reporting/Implementation/FilePathHelper.cs), which is internal so I can't provide a unit test but I can walk through the issue.

**On Windows** let's say the tests are running from `C:\app\tests` and you specify the output path for reporting as `C:\logs\report.txt`.
  - `ReportFileWriter` constructor calls `ReportPathFormatter.ToFormattablePath(@"C:\logs\report.txt")`
  - `ReportPathFormatter.ToFormattablePath` sees no parameters, so no replacement happens.
  - `ReportPathFormatter.ToFormattablePath` calls `FilePathHelper.ResolveAbsolutePath(@"C:\logs\report.txt")`
  - `FilePathHelper.ResolveAbsolutePath` skips the bit where the path starts with `~` and calls `FilePathHelper.Combine(@"C:\app\tests", @"C:\logs\report.txt")`.
  - `FilePathHelper.Combine` returns (simplifying each step)...
    - `Path.Combine(@"C:\app\tests".TrimEnd(Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar), @"C:\logs\report.txt".TrimStart(Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar))`
    - `Path.Combine(@"C:\app\tests".TrimEnd('\\', '/'), @"C:\logs\report.txt".TrimStart('\\', '/'))`
    - `Path.Combine(@"C:\app\tests", @"C:\logs\report.txt")`
    - `@"C:\logs\report.txt"`

So, on Windows, specifying an absolute path works great. However...

**On Linux** let's say the tests are running from `/app/tests` and you specify the output path for reporting as `/logs/report.txt`.
  - `ReportFileWriter` constructor calls `ReportPathFormatter.ToFormattablePath("/logs/report.txt")`
  - `ReportPathFormatter.ToFormattablePath` sees no parameters, so no replacement happens.
  - `ReportPathFormatter.ToFormattablePath` calls `FilePathHelper.ResolveAbsolutePath("/logs/report.txt")`
  - `FilePathHelper.ResolveAbsolutePath` skips the bit where the path starts with `~` and calls `FilePathHelper.Combine("/app/tests", "/logs/report.txt")`.
  - `FilePathHelper.Combine` returns (simplifying each step)...
    - `Path.Combine("/app/tests".TrimEnd(Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar), "/logs/report.txt".TrimStart(Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar))`
    - `Path.Combine("/app/tests".TrimEnd('/', '\\'), "/logs/report.txt".TrimStart('/', '\\'))`
    - `Path.Combine("/app/tests", "logs/report.txt")`
    - `"/app/tests/logs/report.txt"`

Uh oh! By trimming the leading directory separators on the subpath it breaks the ability for Linux to specify an absolute location where reports can be written.

#### Progress

- [x] Feature is implemented,
- [ ] Ensured backward-compatibility,
- [ ] Ensured good debugging experience

**Comment by Suremaker** [comment]
Thank you @tillig for reporting this issue and providing so detailed analysis. I'll take a look and provide fix soon.

**Comment by Suremaker** [comment]
The fix will be included in LightBDD 2.4.2 which I will release around the weekend.

**Comment by tillig** [comment]
Awesome, thanks!
